### ISSUE TITLE
resolved DNS-over-TLS fails with dnsforge.de (i.e. CHACHA20_POLY1305_SHA256 is not supported by resolved/gcrypt)

### ISSUE THREAD

**Issue by CoRoe** [systemd/systemd#25894]
### systemd version the issue has been seen with

245.4-4ubuntu3.19

### Used distribution

Linux Mint

### Linux kernel version used

5.4.0-135-generic

### CPU architectures issue was seen on

x86_64

### Component

systemd-resolved

### Expected behaviour you didn't see

systemd-resolved establishes a TLS session with the recursive resolver dnsforge.de (176.9.93.198).

### Unexpected behaviour you saw

TLS session establishment between systemd-resolved and dnsforge.de fails after several _client-hello_ -- _change cipher spec_ exchanges.

### Behaviour with other DNS resolvers:

Google (8.8.8.8) OK
Cloudflare (1.1.1.1) OK
Quad9 (9.9.9.9) needs several retries and succeeds after some seconds

### Steps to reproduce the problem

- Configure in /etc/systemd/resolved.conf: DNS=176.9.93.198
- Restart systemd-resolved.service
- Try to resolve a hostname

[pcap traces.zip](https://github.com/systemd/systemd/files/10325607/pcap.traces.zip)
[resolver-conf.zip](https://github.com/systemd/systemd/files/10325619/resolver-conf.zip)

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
$ dig www.spiegel.de

; <<>> DiG 9.16.1-Ubuntu <<>> www.spiegel.de
;; global options: +cmd
;; connection timed out; no servers could be reached
```

**Comment by Winterhuman** [comment]
Have you tried `DNS=176.9.93.198#dnsforge.de`?

**Comment by CoRoe** [comment]
Yes, I had tried, and lookup fails (see the attached pcap files).

I looked a bit deeper:

```
| Server     | Cipher suite selected by server | Result    |
|------------+---------------------------------+-----------|
| dnsforge   | 0x1303 CHACHA20_POLY1305_SHA256 | Fail      |
| Google     | 0x1302 AES_256_GCM_SHA384       | OK        |
| Cloudflare | 0x1302 AES_256_GCM_SHA384       | OK        |
| Quad9      | 0x1302 AES_256_GCM_SHA384       | (Unclear) |
```
Possible cause of the failure: _systemd-resolved_ accepts cipher suite CHACHA20_POLY1305_SHA256 but doesn't really support it.

**Comment by CoRoe** [comment]
The debug log says:

Jan 02 11:22:18 klapp systemd-resolved[47146]: Cache miss for push.services.mozilla.com IN A
Jan 02 11:22:18 klapp systemd-resolved[47146]: Transaction 63898 for <push.services.mozilla.com IN A> scope dns on */*.
Jan 02 11:22:18 klapp systemd-resolved[47146]: Using feature level TLS+EDNS0 for transaction 63898.
Jan 02 11:22:18 klapp systemd-resolved[47146]: Sending query via TCP since UDP isn't supported.
Jan 02 11:22:18 klapp systemd-resolved[47146]: Using feature level TLS+EDNS0 for transaction 63898.
Jan 02 11:22:18 klapp systemd-resolved[47146]: Failed to invoke gnutls_handshake: **Error in the certificate verification**.
Jan 02 11:22:18 klapp systemd-resolved[47146]: Connection failure for DNS TCP stream: Connection refused
